Sources/OpenAPIRuntime/Interface/ErrorHandlingMiddleware.swift

```diff
@@ -0,0 +1,86 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftOpenAPIGenerator open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftOpenAPIGenerator project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftOpenAPIGenerator project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import HTTPTypes
+
+/// Error Handling middleware that converts an error to a HTTP response
+///
+/// This is an opt-in middleware that adopters may choose to include to convert application errors
+/// to a HTTP Response
+///
+/// Inclusion of this ErrorHandling Middleware should be accompanied by confirming errors to
+/// HTTPResponseConvertible protocol. Only errors  confirming to HTTPResponseConvertible are converted to a HTTP response. Other errors are re-thrown from this middleware.
+///
+/// Example usage
+/// 1. Create an error type that conforms to HTTPResponseConvertible protocol
+/// ```swift
+/// extension MyAppError: HTTPResponseConvertible {
+///    var httpStatus: HTTPResponse.Status {
+///    switch self {
+///        case .invalidInputFormat:
+///            .badRequest
+///        case .authorizationError:
+///            .forbidden
+///        }
+///    }
+/// }
+/// ```
+///
+/// 2. Opt in to the error middleware while registering the handler
+///
+/// ```swift
+/// let handler = try await RequestHandler()
+/// try handler.registerHandlers(on: transport, middlewares: [ErrorHandlingMiddleware()])
+///
+
+public struct ErrorHandlingMiddleware: ServerMiddleware {
```

Comment:
Thank you so much for adding documentation comments; we really value them! Do you mind polishing them up a little, though?

- Ending sentences with punctuation: either full stop or colon, depending on context.
- Reflowing lines at a consistent width.
- The second sentence seems a little redundant given the headline comment.
- In the third, can we remove the double-space in the middle of the sentence, and use double-backticks to link to the referred documentation of `HTTPResponseConvertible`?
- Terminate the final code-fence.
- Remove any empty comment lines at the end of the doc comment block.
- Remove any empty lines between the comment block and the symbol to which it should be attached.

This feedback applies to any other documentation comments too.

Reply:
Separately, should we add documentation about the importance of its position in the responder chain?